.github/workflows/gh-page.yml

```diff
@@ -0,0 +1,49 @@
+name: Deploy Specifications & Docs to GitHub Pages
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  release:
+    name: GitHub Pages
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Setup OCaml (because of ocaml-gen)
+        run: sudo apt update && sudo apt install ocaml
+
+      - name: Build Rust Documentation
+        run: |
+          rustup install nightly
+          RUSTDOCFLAGS="--enable-index-page -Zunstable-options" cargo +nightly doc --all --no-deps
+
+      - name: Check that up-to-date specification is checked in
+        run: |
+          cargo install cargo-spec
+          cd book/specifications
+          cd kimchi && make build && cd ..
+          cd poly-commitment && make build && cd ..
+          git diff --exit-code
```

Comment:
this makes sure that if someone changes the specifications, they have also run `cargo-spec` and have staged the newly generated specs

Reply:
yeah I think that's a good idea!